### ISSUE TITLE
OrthographicViewport zooming alignment issue 

### ISSUE THREAD

**Issue by gnavvy** [uber/deck.gl#1252]
currently, scaling while keeping the center position when initiating a new OrthographicsViewport will lead to an offset (top-left alignment), but no zooming. --> behave differently as before.

Expected result for orthographic zooming: (result from v4.x, center aligned)

![4.x](https://user-images.githubusercontent.com/144743/34231176-cd3b76f4-e58f-11e7-8128-da8873c1430b.gif)

Current result from v5.0-beta: (top left aligned with wrong scaling)

![5.x](https://user-images.githubusercontent.com/144743/34231178-ceec6738-e58f-11e7-8e25-d3154b962651.gif)

will create a standalone test example to double check if this issue comes from deck.gl or it's in the app.

**Comment by gnavvy** [comment]
narrowed down to [913](https://github.com/uber/deck.gl/pull/913/files).

**Comment by 1chandu** [comment]
Looking at the above gifs, it looks like it is scaling down gl-viewport (screen space rect), instead of stretching.

**Comment by ibgreen** [comment]
What happened in #913  was that the project utility functions had been moved to viewport-mercator-project. So now we could delete them from deck.gl, import them from vpm, and avoid some code duplication.

Some function may have changed in viewport-mercator-project?

One possible thing to look at is that some of the project functions have a parameter that indicates whether pixels are in "top left" (CSS) or "bottom left" (WebGL) coordinates.

**Comment by 1chandu** [comment]
I was able to reduce this to alpha.6 (9c541d3, #899 **GOOD**) and alpha.7 (ade0646, #913 **BAD**), but couldn't get the build to work to narrow it further down to single PR.

**Comment by gnavvy** [comment]
This issue can be bypassed without changing the current API and implementation.
An example was added in https://github.com/uber/deck.gl/pull/1266.

I was not able to locate where exactly that changed the behavior, but before we add multiple viewport support, the width and height of the viewport and canvas are always in sync (referring to the same width and height). Now they are not and hence the content in the viewport - when scaled - are not strengthened to fill the whole canvas, led to result as shown in the second gif.

In short: (already included in the doc.)

1. when using ortho-viewport without zooming, just specify width, height, left, and top. The ortho-viewport class will assume viewport size and canvas size are the same and derive the right and bottom internally.

2. when using ortho-viewport with zooming, specify width, height (for canvas), left, top, **and right bottom** (for viewport) to achieve the zooming effect.

Will close this issue for now.
